code-challenges.test.js

```diff
@@ -38,14 +66,55 @@ const votes1 = { upVotes: 13, downVotes: 2 }
 const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
 
+describe('tally', () => {
+  it('return end tally', () => {
+    expect(tally(votes1)).toEqual(11)
+    expect(tally(votes2)).toEqual(-31)
+  })
+})
+
 // b) Create the function that makes the test pass.
 
+// Begin the tally function, which takes an object as input.
+// Access the upVotes and downVotes properties of the object.
+// Subtract the value of downVotes from the value of upVotes.
+// Return the result of the subtraction as the tally.
+
+const tally = (obj) => {
+  return obj.upVotes - obj.downVotes
+}
+
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const dataArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+const dataArray1 = ['array', 'object', 'number', 'string', 'Boolean']
+const dataArray2 = ['string', 'null', 'Boolean', 'string', 'undefined']
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
+describe('removeDups', () => {
+  it('return end tally', () => {
+    expect(removeDups(dataArray1, dataArray2)).toEqual([
+      'array',
+      'object',
+      'number',
+      'string',
+      'Boolean',
+      'null',
+      'undefined',
+    ])
+  })
+})
+
 // b) Create the function that makes the test pass.
+
+// Begin the removeDups function, which takes any number of arrays as input using the rest parameter syntax.
+// Use the reduce method on the input arrays to flatten them into a single array. Start with an empty array as the initial value of the accumulator.
+// Use the Set constructor and the Array.from method to create a new array with duplicate elements removed.
+// Return the new array as the result of the function.
+
+const removeDups = (...arrays) => {
+  return Array.from(
+    new Set(arrays.reduce((acc, curr) => [...acc, ...curr], []))
+  )
+}
```

Comment:
Crushed that stretch challenge. Great use of your tools, and so far the only solution I've seen to use `.reduce` for this problem. Nice work!